Fortnox.NET/WebSockets/FortnoxWebSocketClient.cs

```diff
@@ -107,161 +92,89 @@ private async Task<T> SendAndRecieveOnce<T>(string message)
-        }
-
-        /// <summary>
-        /// Initiates the connection to the underlying <see cref="System.Net.WebSockets.ClientWebSocket"/> as well as subscribes to all the specified topics as an asynchronous operation.
-        /// </summary>
-        /// <returns>The current instance of <see cref="FortnoxWebSocketClient"/> as an asynchronous operation.</returns>
-        public async Task<FortnoxWebSocketClient> Connect()
-        {
-            // TODO(Oskar): Should we keep anonymous objects or create something concrete?
-            await _webSocket.ConnectAsync(_fortnoxWebSocketURI, CancellationToken.None);
-
-            var AddTenantsObject = new { command = WebSocketCommands.AddTenants, clientSecret = _clientSecret, accessTokens = new List<string>(_accessTokens) };
-            var AddTenantsResult = await SendAndRecieveOnce<WebSocketCommandResponse>(JsonConvert.SerializeObject(AddTenantsObject));
-
-            if (AddTenantsResult.Result.Equals("error"))
-            {
-                throw new WebSocketException("Unable to add tenants.");
-            }
-
-            if (_topics.Count <= 0)
+            var deserializedResult = JsonConvert.DeserializeObject<WebSocketResponse>(resultString, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
+            if (deserializedResult.TenantId.HasValue)
             {
-                throw new WebSocketException("No topics specified.");
+                deserializedResult.Type = WebSocketResponseType.EventResponse;
             }
 
-            var AddTopicObject = new { command = WebSocketCommands.AddTopics, topics = new List<object>() };
-            foreach(var topic in _topics)
+            if (deserializedResult.Type == WebSocketResponseType.CommandResponse)
             {
-                AddTopicObject.topics.Add(new { topic = topic });
+                if (deserializedResult.Result.Equals("ok") && deserializedResult.Response.Equals(WebSocketCommands.AddTenants))
+                {
+                    foreach (var item in deserializedResult.TenantIDs)
+                    {
+                        AccessTokenTenantId.Add(item.Key, item.Value);
+                    }
+                }
             }
 
-            var AddTopicResult = await SendAndRecieveOnce<WebSocketCommandResponse>(JsonConvert.SerializeObject(AddTopicObject));
-            if (AddTopicResult.Result.Equals("error"))
-            {
-                throw new WebSocketException($"Error adding topic: {string.Join(",", AddTopicResult.InvalidTopics)}");
-            }
+            return deserializedResult;
+        }
 
-            var SubscribeObject = new { command = WebSocketCommands.Subscribe };
-            var SubscribeResult = await SendAndRecieveOnce<WebSocketCommandResponse>(JsonConvert.SerializeObject(SubscribeObject));
+        public async Task AddTenant(string accessToken)
+        {
+            var AddTenantsObject = new 
```

Comment:
Can be done using a initilaze list:

```CSharp
var AddTenantsObject = new 
            { 
                command = WebSocketCommands.AddTenants, 
                clientSecret = _clientSecret, 
                accessTokens = new List<string>() { accessToken }
            };
```

See [this](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers#collection-initializers)